src/ray/id.cc

```diff
@@ -24,18 +24,26 @@ std::mt19937 RandomlySeededMersenneTwister() {
   return seeded_engine;
 }
 
+uint64_t MurmurHash64A(const void *key, int len, unsigned int seed);
+
 UniqueID::UniqueID() {
   // Set the ID to nil.
   std::fill_n(id_, kUniqueIDSize, 255);
+  hash_ = MurmurHash64A(&id_[0], kUniqueIDSize, 0);
+}
+
+UniqueID::UniqueID(const std::string &binary) {
+  std::memcpy(&id_, binary.data(), kUniqueIDSize);
+  hash_ = MurmurHash64A(&id_[0], kUniqueIDSize, 0);
 }
 
 UniqueID::UniqueID(const plasma::UniqueID &from) {
   std::memcpy(&id_, from.data(), kUniqueIDSize);
+  hash_ = MurmurHash64A(&id_[0], kUniqueIDSize, 0);
 }
 
 UniqueID UniqueID::from_random() {
-  UniqueID id;
-  uint8_t *data = id.mutable_data();
+  std::string data;
```

Comment:
Will `std::string data(kUniqueIDSize, 0);` and `data[i] = ` has better performance than `data.push_back`?